python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -74,3 +79,158 @@ def deserialize_partition_from_db_string(
     if partition.startswith("["):
         return deserialize_as(partition, MultiDimensionalPartitionKey)
     return partition
+
+
+class MultiDimensionalPartition(Partition):
```

Comment:
I'm not entirely convinced that this needs to be its own class. If I understand correctly, all it does is add an alias to the value called `partitions_by_dimension`. Thoughts on just using vanilla `Partition`s?